src/metrics/export/serverexporter/serverexporter.go

```diff
@@ -0,0 +1,117 @@
+package serverexporter
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	pb "github.com/protolambda/rumor/proto"
+	"google.golang.org/grpc"
+)
+
+// TODO:
+//      - Aggregate a Context framework to shut down the endless loop
+//      - Choose the initialization parameteres / place in the code
+//      - Wait untill the gRPC format is defined
+
+type ServerEndpoint struct {
+	IP   string
+	Port string
+	// Notification channels
+	NewPeerDiscov   chan *pb.NewPeerMetadataRequest
+	NewPeerConn     chan *pb.NewPeerMetadataRequest
+	NewPeerDisc     chan *pb.NewPeerMetadataRequest
+	NewPeerMetadata chan *pb.NewPeerMetadataRequest
+	close           chan struct{}
+}
+
+// Initialize the Peer Metrics Export for towards the Armiarma Server Endpoint
+func NewServerExport(ip string, port string) *ServerEndpoint {
+	se := &ServerEndpoint{
+		IP:              ip,
+		Port:            port,
+		NewPeerDiscov:   make(chan *pb.NewPeerMetadataRequest, 200),
+		NewPeerConn:     make(chan *pb.NewPeerMetadataRequest, 200),
+		NewPeerDisc:     make(chan *pb.NewPeerMetadataRequest, 200),
+		NewPeerMetadata: make(chan *pb.NewPeerMetadataRequest, 200),
+		close:           make(chan struct{}),
+	}
+	return se
+}
+
+// Function that runs the routiene to export the data to the Armiarma server
+func (c *ServerEndpoint) Start() {
+	fmt.Println("Starting Server Export towards Armiarma Server: ", c.IP, ":", c.Port)
+	// (generate the gRPC host and configure it)
+	serverAddr := c.IP + ":" + c.Port
+	conn, err := grpc.Dial(serverAddr, grpc.WithInsecure(), grpc.WithBlock())
+	if err != nil {
+		fmt.Println("connection with the Armiarma Server couldn't be made")
+	}
+	// keep the gRPC method
+	client := pb.NewAggregatorClient(conn)
+	// Call the Export Infinite Loop
+	go c.exportLoop(conn, &client)
+}
+
+// Close/Stop fucntion to finish the end loop to export the metadata
+func (c *ServerEndpoint) Stop() {
+	fmt.Println("sending closing signal to end the server exporter loop")
+	<-c.close
+}
+
+// Main loop where the connection with the Armiarma Server will be orchestrated
+func (c *ServerEndpoint) exportLoop(conn *grpc.ClientConn, client *pb.AggregatorClient) {
+	// Initialize and blablabla
+	for {
+		// Infinite loop that will receive each of the events and act accordingly
+		select {
+		case msg := <-c.NewPeerDiscov:
```

Comment:
I was thinking that perhaps we should merge `NewPeerDiscovered` with `NewPeerMetadata`, as the latter is a super set of the first one. I mean, both of them share the same data (ip, locations, etc) but `NewPeerMetadata` contains more information. So perhaps both can share the same rpc call and structs, so we can kill two birds with one stone. What do you think?

Reply:
Here I might disagree a bit with the merge of these two metrics. I'm not familiar with the optimal setup of gRPCs, is better to send many little messages or fewer bigger messages. I would suggest analyzing/testing which case performs better with our tool and merge or separate message content based on the results

Reply:
Absolutely. Lets focus by now on a higher level and then we can fine tune it. Having both `NewPeerDiscovered` and `NewPeerMetadata` by now is fine by me. Now that I think it twice it also makes sense from a conceptual point of view.